### ISSUE TITLE
Get rid of the book identification

### ISSUE THREAD

**Issue by Coolthulhu** [CleverRaven/Cataclysm-DDA#16085]
Our book identification is a good example on how to not do item identification in a roguelike:
- It hides information that the player can memorize, presenting exactly zero challenge to a player who keeps notes/has good memory
- The only kind of information it hides that actually can matter is whether the standpipe log is a regular standpipe log, or the weird one that has elf-a mutagen. And even it only means that you may need to identify standpipe log twice (once you id the regular one, you can tell the elf-a one apart at a glance). This whole elf-a standpipe thing itself is stupid anyway, worth getting rid of on its own.
- It is annoying, because it requires identifying every found book to use its recipes. Reading the book once takes a tiny amount of character time, but requires going through menus and shuffling items around since you can't read from the ground (you can pickup-and-read from the ground, but that requires shuffling items around)
- It is annoying once again, because if you don't id all the books (which takes a negligible amount of time), you won't get book coloration for convenience in looting and picking books to read

As for redeeming factors:
- Books that aren't id'd are most likely the books you don't have yet, making it a tiny bit less likely that you'll grab duplicates.

And that's it.

I'd say it's deep into "not worth not deleting" territory.

Item ID makes sense when it forces the player to make decisions, such as in "should I risk using this potion to heal poison, hoping it's healing and not more poison?" or "I need a helmet but this one can be cursed". Skimming a book that isn't Necronomicon isn't dangerous, it doesn't require a decision.

**Comment by kevingranade** [comment]
The primary rationale for it to me is the simple fact that otherwise the
very first time a character sees a book from across a room, they know all
of its contents.  This is a simply absurd situation and well worth
preventing.
If you have an alternate solution for this absurdity, I'd love to hear it,
but I'm strongly against reintroducing it.

Fundamentally, I don't like magical character knowledge. Most games make
this compromise for reasons similar to what you outline, and IMO they are
far worse for it.  The acquisition of books is a very important part of
character progression in dda, and it deserves to not be glossed over.

P.s. The standpipe log can go.

**Comment by Labtop-215** [comment]
Not grabbing duplicates is a good thing.  Also, the book titles are pretty explanatory anyway as for what they are for.  The information they do hide isn't really worth memorizing, and memorizing it isn't really worth doing since you arn't much better off than guessing anyway, unless you are after a specific recipe for mutagen.

Some examples...  Under the Hood, obviously the hood of a car, so mechanics.
Chemical Reference...  Cooking.
Knife Fighters notes...  Cutting Weapons Skill
Tao of the Handgun...  Handgun Skill
FM 23-16 Army Marksmanship Manual... Marksmanship
Building 101...  Construction

Maybe some of the more valuable books shouldn't have their titles so prominently displayed and visible until you read them?  It would make looting books in low light and dangerous conditions more interesting at least (or frustrating, depending on your outlook), and it would subtlety nerf books a little bit since book acquisition seems to be one of the few things that stand in the way of from you becoming a demigod of space time.

**Comment by warmist** [comment]
I would suggest randomly generating names maybe? Currently there is little "content" proc gen in CDDA. This would be like potions in other roguelikes. This also nicely ties in to the quite powerful nature of books: they teach you skills AND recipes.

**Comment by illi-kun** [comment]
> The primary rationale for it to me is the simple fact that otherwise the very first time a character sees a book from across a room, they know all of its contents.

So all we need is two levels (or more, if needed) of item info:
1) general item description: something general and obvious: "this is a red book" (available by default from any distance)
2) fine detailed item description: current description of item + receipts in case of books (available for distance <= 1, or if it's in your inventory).

**Comment by Coolthulhu** [comment]
>  The information they do hide isn't really worth memorizing

Then it's not worth hiding either. And removing the annoyance of skimming is worth removing hiding.

There are 2-3 low level crafting books. Only one of them has duct tape recipe. There are 2-3 low level first aid books with very similar names, but different levels.

> So all we need is two levels (or more, if needed) of item info

That sounds really annoying for anyone with a base bigger than 3x3. Unless you want to note all identified objects (object types?) somehow, to prevent player forgetting everything about the stash after taking a single step away from it.

Plus, it doesn't really solve the problem unless it also hides all the other ways of eliminating trash books - volume, weight, color etc.

> The primary rationale for it to me is the simple fact that otherwise the very first time a character sees a book from across a room, they know all of its contents.

Assume the character skims it as soon as possible instead of requiring the player to manually order it.

> This is a simply absurd situation and well worth preventing.

Current solution replaces it with even worse thing: requiring meta-knowledge and punishing the player with tedium for the lack of it. Then punishing those with the meta-knowledge with less tedium too, just to maintain the identification mechanic.

The character will still beeline for the correct book because the player knows which one is it, because the book selection is fixed and small.

---

Basically, the only way to make item identification actually make sense is to make books randomly generated in a way that doesn't allow using volume, weight, color and price for instant identification.

Maybe add a "recipe mask" to make different "editions" of the book have less recipes, to make getting duplicates make sense or something. Or even better: generate recipe lists.

**Comment by vache** [comment]
I actually like the standpipe log thing.  The idea to me is that someone hid a piece of valuable information in something otherwise innocuous.  It adds a tiny little bit of atmosphere.

**Comment by kevingranade** [comment]
On Apr 7, 2016 8:23 AM, "Coolthulhu" <notifications@github.com> wrote:
>>
>> The information they do hide isn't really worth memorizing
>
> Then it's not worth hiding either.

Hiding the info isn't justified based on how valuable the information is,
it's justified based on the fact that the player has no way to know it.
Also it in theory would make looting books a little more dangerous due to
the time cost, but there are no enemies that capitalize on that at the
moment.

>> So all we need is two levels (or more, if needed) of item info

There's no reason to regress to forgetting what you know about the book
once that knowledge is acquired.

> Assume the character skims it as soon as
> possible instead of requiring the player to
> manually order it.

So either picking up a book takes several minutes if you haven't skimmed it
before, or merely touching a book tells you it's contents. These are both
extremely unreasonable.

> Current solution replaces it with even worse
> thing: requiring meta-knowledge and
> punishing the player with tedium for the lack
> of it. Then punishing those with the
> meta-knowledge with less tedium too, just to
> maintain the identification mechanic.

I find it hard to justify calling it 'punishing' to require a single item
activation to learn all this information about a book. If the ui needs
streamlining, sure let's look at that, but I'm just not seeing the huge
time cost you seem to be reacting to.

> The character will still beeline for the correct > book because the
player knows which one is > it, because the book selection is fixed and
small.
>
> ________________________________
>
> Basically, the only way to make item
> identification actually make sense is to make
> books randomly generated in a way that
> doesn't allow using volume, weight, color and
> price for instant identification.
>

More selection would be fine, making the description slightly more generic
would be fine (the only rationale I know of for books having different
colors is exactly this kind of metagaming, and most books should have
similar weights, at least to the kg or lb), randomization not so much.

If someone wants to spoil their game by looking up or memorizing game
facts, they can, but i cant do anything to improve their game experience.
I'm focusing on the players that don't do that.

**Comment by RogueYun** [comment]
I'm just brainstorming so please forgive me if some of my suggestions are not viable... My hope is that someone might gain some insight for a better idea.

Maybe there can be books with standard stuff where the title pretty much says what is in the book AND a bunch of other "Journals" that contain random amounts of information. "This survivor appears to have been a <low/medium/high> level <occupation>! There is some information here on how he used this knowledge for practical use before his inevitable demise." At which point the name would change to something like "Survivor <occupation>'s Journal" and then maybe you could have a random skill level bonus and read time requirement (maybe the read time is based on his handwriting?). You could also make them "entertaining", "factual", or "boring". Something along those lines to give a random morality boost/decrease. 

If this is the case, I'd like for renamed title of the book to have the level requirement listed as well maybe in brackets? something like "A <occupation>'s Journal [<min level>|<max level>|<read time>|<morality boost>]". It would make sorting/using books a lot easier!

Maybe they could be considered random articles in a magazine. So the name of the magazine doesn't give you any hint. Something like "Bodacious Babes of Guatamala". But when it is read it says, "This magazine contains a very interesting article on <random skill associated action> and some recipes to boot!" Maybe some books could be done like that as well?

Perhaps books can be stored on unmarked flash drives (kindle like books?) or CD's (audio books?) for use with working terminals or PDAs?

You could add some sub history to books becoming obsolete? Perhaps recipes are more separate from books and you find them in survivor notes or torn out book pages? Maybe this could be explained historically by some sort of mass book burning? Seems far fetched but I'm just throwing out ideas.

**Comment by BFG1992** [comment]
Just my two cents, but still...

I like current mechanics of book identification. It just makes sense, so I've never thought of it as of something annoying or tedious.
Except one thing. As **Coolthulhu** said:

> There are 2-3 low level crafting books. Only one of them has duct tape recipe. There are 2-3 low level first aid books with very similar names, but different levels.

This. I often have a situation when my survivor have two similar books and want to know which one has more recipes in it, and exactly what recipes are absent (or present) in other book, to throw one out, because of free space management and just to have fun neatly organizing stuff.
We have the compare menu. So, how about making it useful to compare recipes in different books? I think that this would be great.

**Comment by Chase-san** [comment]
#6085 touches on this somewhat with a visual description for books, but extends it to other things as well (at the bottom).

**Comment by kevingranade** [comment]
The proposal as stated (remove book id) isn't happening, any number of things can be done to streamline it, but removing it entirely is simply not an option.
